com.unity.shadergraph/Editor/Terrain/TerrainSubTarget.cs

```diff
@@ -0,0 +1,442 @@
+            {
+                displayName = "Terrain Instanced Per Pixel Normal",
+                referenceName = "_TERRAIN_INSTANCED_PERPIXEL_NORMAL",
+                type = KeywordType.Boolean,
+                definition = KeywordDefinition.ShaderFeature,
+                scope = KeywordScope.Local,
+            };
+
+            public static KeywordCollection MainKeywords = new KeywordCollection
+            {
+                { BlendHeight },
+                { InstancedPerPixelNormal },
+                { MaskMap },
+                { NormalMap },
+            };
+
+            public static KeywordCollection BasemapGenKeywords = new KeywordCollection
+            {
+                { NormalMap },
+                { MaskMap },
+                { BlendHeight },
+            };
+        }
+
+        public static class Includes
+        {
+            public static string kTerrainProps = "Packages/com.unity.render-pipelines.core/ShaderLibrary/Terrain/TerrainProps.hlsl";
+            public static string kTerrainSplatmap = "Packages/com.unity.render-pipelines.core/ShaderLibrary/Terrain/TerrainLayer_Includes.hlsl";
+
+            public static IncludeCollection TerrainIncludes = new IncludeCollection()
+            {
+                {   kTerrainProps, IncludeLocation.Pregraph },
+                {   kTerrainSplatmap, IncludeLocation.Pregraph },
+            };
+        }
+
+        public enum TerrainShaders
+        {
+            Main = 0,
+            BasemapGen = 1,
+            Basemap = 2,
+            Add = 3,
+            Count = 4,
+        }
+
+        private static string[] TerrainShaderSuffix = new string[4]
+        {
+            "",
+            "_BasemapGen",
+            "_Basemap",
+            "_Add",
+        };
+
+        private static string[] DependencyNames = new string[4]
+        {
+            "",
+            "BaseMapGenShader",
+            "BaseMapShader",
+            "AddPassShader",
+        };
+
+        public static string GetDependencyName(TerrainShaders terrainShader)
+        {
+            return DependencyNames[Mathf.Clamp((int)terrainShader, 0, 3)];
+        }
+
+        public static string GetTerrainShaderName(string shaderName, TerrainShaders terrainShader)
+        {
+            return shaderName + TerrainShaderSuffix[Mathf.Clamp((int)terrainShader, 0, 3)];
+        }
+
+        public static string GetTerrainShaderName(string shaderName, string dependencyName)
+        {
+            return shaderName + TerrainShaderSuffix[Mathf.Clamp(GetTerrainShaderId(dependencyName), 0, 3)];
+        }
+
+        private static int GetTerrainShaderId(string dependencyName)
+        {
+            for (int i = 0; i < (int)TerrainShaders.Count; i++)
+            {
+                if (dependencyName == DependencyNames[i])
+                    return i;
+            }
+            return -1;
+        }
+
+        public static void SetDstBlendValue(string dstBlendPropName)
+        {
```

Comment:
Since URP and HDRP have different property names used to set DstBlend, this is how they can configure them